lambda/dockerimagebuilder/__init__.py

```diff
@@ -53,20 +54,38 @@ def handler(event: Dict[str, Any], context) -> None:  # type: ignore [no-untyped
     rendered_userdata = rendered_userdata.replace("{{BASE_IMAGE}}", shlex.quote(base_image))
     rendered_userdata = rendered_userdata.replace("{{MOUNTS3_DEB_URL}}", shlex.quote(mounts3_deb_url))
     rendered_userdata = rendered_userdata.replace("{{ECR_URI}}", shlex.quote(os.environ["LISA_ECR_URI"]))
-    rendered_userdata = rendered_userdata.replace("{{IMAGE_ID}}", str(uuid.uuid4()))
+    rendered_userdata = rendered_userdata.replace("{{IMAGE_ID}}", image_tag)
 
     print("Creating instance with userdata: ")
     print(rendered_userdata)
     try:
-        ec2_resource.create_instances(
+        instances = ec2_resource.create_instances(
             ImageId=ami_id,
             MinCount=1,
             MaxCount=1,
             InstanceType="m5.large",
             UserData=rendered_userdata,
             IamInstanceProfile={"Arn": os.environ["LISA_INSTANCE_PROFILE"]},
             BlockDeviceMappings=[{"DeviceName": "/dev/xvda", "Ebs": {"VolumeSize": 32}}],
```

Comment:
does the block device get deleted on termination too? just want to confirm that while we're here

Reply:
yep just confirmed